### PR TITLE
Mars Protocol: Simplify the way of fetching TVL 

### PR DESCRIPTION
**Pull request by linkielink**
This PR simplifies fetching the TVL for each chain by querying the total deposit amount for each asset in the chain's params contract.

### CODE REVIEW

**Comment by llamatester** [comment]
Error while running adapter at projects/mars: 
        
 
  
    
    
    TypeError: Cannot read properties of undefined (reading 'params')
    Truncated error stack:
    
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/mars/index.js:69:34
        at Array.map (<anonymous>)
        at addCoinsFromAssetParams (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/mars/index.js:67:33)
        at fetchDeposits (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/mars/index.js:55:11)
        at async neutronTVL (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/mars/index.js:28:3)
        at async getTvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:52:23)
        at async /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:145:9
        at async Promise.all (index 0)
        at async Promise.all (index 3)

**Comment by llamatester** [comment]
The adapter at projects/mars exports TVL: 
        
 
 
    osmosis                   18.92 M
    neutron                   2.63 M
    terra                     0
    
    total                    21.55 M

**Comment by llamatester** [comment]
Error while running adapter at projects/mars: 
        
 
  
    
    
    Error: Failed to get https://rest-kralum.neutron-1.neutron.org/cosmwasm/wasm/v1/contract/osmo1nlmdxt9ctql2jr47qd4fpgzg84cjswxyw6q99u4y4u4q6c2f5ksq7ysent/smart/eyJhbGxfYXNzZXRfcGFyYW1zIjp7ImxpbWl0Ijo1LCJzdGFydF9hZnRlciI6bnVsbH19
    Truncated error stack:
    
        at get (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/helper/http.js:33:11)
        at async queryContract (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/helper/chain/cosmos.js:196:5)
        at async fetchDepositsMinusDebts (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/mars/index.js:47:25)
        at async neutronTVL (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/mars/index.js:32:3)
        at async getTvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:52:23)
        at async /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:145:9
        at async Promise.all (index 0)
        at async Promise.all (index 3)
        at async /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:188:3

**Comment by llamatester** [comment]
The adapter at projects/mars exports TVL: 
        
 
 
    osmosis                   18.06 M
    neutron                   2.66 M
    terra                     0
    
    total                    20.72 M

**Comment by llamatester** [comment]
The adapter at projects/mars exports TVL: 
        
 
 
    osmosis                   18.06 M
    neutron                   2.66 M
    terra                     0
    
    total                    20.72 M

**Comment by llamatester** [comment]
The adapter at projects/mars exports TVL: 
        
 
 
    osmosis                   13.31 M
    neutron                   2.04 M
    terra                     0
    
    total                    15.35 M

File: projects/mars/index.js [hunk 1]
```diff
@@ -1,71 +1,81 @@
...
-  return balance.balance.amount;
+async function addCoinsFromAssetParams(coins, assetParams, chain) {
+  const assetDenoms = assetParams.map(asset => asset.denom);
+
+  // query the deposited amount for each asset and add it to the coins array
+  await Promise.all(assetDenoms.map(async denom => {
+    let totalDepositInfo = await queryContract({
+      contract: paramsContractAddresses[chain],
+      chain,
+      data: { 'total_deposit': {
```
[the remark this answers is not in the document]
Comment (linkielink) [reply projects/mars/index.js:99]: @g1nt0ki, now it does; thanks for pointing that out. I forgot that TVL is deposits - borrowed values.

It fetches the debt amount of every market now and deducts that from the deposits. Excuse the commit spam, but my local tests weren't returning any data (I have to fix that when I can).

**Comment by g1nt0ki** [comment]
@linkielink thanks for the PR, I have merged it with a minor fix

**Pull request closed without merging**